### PR TITLE
Implement bitcast for 66-bit date type

### LINKED ISSUE #5470: Bug on supporting bit_cast<uint64_t>(double)
Following code will cause internal error:
```
RWStructuredBuffer<uint64_t> out;

uint64_t cast(double x)
{
    return bit_cast<uint64_t>(x);
}

[shader("compute")]
[numthreads(1,1,1)]
void computeMain()
{
    double x = 1.0;

    out[0] = cast(x);
}
```

note, there is no problem just doing 
```
  out[0] = bit_cast<uint64_t>(1.0lf);
```

### PR DESCRIPTION
**Pull request by kaizhangNV**
Close #5470

### CODE REVIEW

File: source/slang/slang-emit-hlsl.cpp [hunk 1]
```diff
@@ -11,6 +11,23 @@
 namespace Slang
 {
 
+static const char* kHLSLBuiltInPrelude64BitCast = R"(
+uint64_t _slang_bitcast64(double x)
```
Comment (csyonghe) [inline source/slang/slang-emit-hlsl.cpp:15]: Should be named to _slang_asuint64() to align with existing HLSL intrinsic

File: source/slang/slang-emit-hlsl.cpp [hunk 2]
```diff
@@ -11,6 +11,23 @@
...
+{
+    uint32_t low;
+    uint32_t high;
+    asuint(x, low, high);
+    return ((uint64_t)high << 32) | low;
+}
+
+double _slang_bitcast64(uint64_t x)
```
Comment (csyonghe) [inline source/slang/slang-emit-hlsl.cpp:23]: _slang_asdouble()

File: source/slang/slang-emit-hlsl.h [hunk 3]
```diff
@@ -28,6 +28,9 @@ class HLSLSourceEmitter : public CLikeSourceEmitter
 
 protected:
     RefPtr<HLSLExtensionTracker> m_extensionTracker;
+    Dictionary<const char*, IRStringLit*> m_builtinPreludes;
```
[the remark this answers is not in the document]
Comment (kaizhangNV) [reply source/slang/slang-emit-hlsl.h:31]: sure, will move that.

File: source/slang/slang-emit-hlsl.cpp (hunk 1 above)
Comment (kaizhangNV) [reply source/slang/slang-emit-hlsl.cpp:15]: ack, will update

**Comment by kaizhangNV** [comment]
@csyonghe can you give it another review?

**Review by csyonghe — APPROVED** [review]
LGTM.

**Pull request merged**